src/dataflow/src/source/mod.rs

```diff
@@ -1191,6 +1171,14 @@ where
                 session.give_iterator(ts_bindings_retractions);
             }
 
+            // Maintain a capability set that tracks the durability frontier.
+            let durability_frontier = timestamp_histories.durability_frontier();
+            cap.downgrade(durability_frontier.clone());
```

Comment:
I finally understand the issue here: if you keep a single capability set that tracks the durability frontier, then you can't (at least in the way the code is currently set up) send data at times less than the durability frontier

Reply:
wheres having two separate capabilities is a way to track the min between "what data have i sent sent out" vs "what data has been persisted"